it-tests/BasicFlow.test.ts

```diff
@@ -0,0 +1,51 @@
+import { By, CustomEditor, EditorView,  until,  VSBrowser, WebDriver, WebView } from 'vscode-extension-tester';
+import { assert } from 'chai';
+import * as path from 'path';
+
+describe('Kaoto basic development flow', function () {
+	this.timeout(120000);
+
+	const workspaceFolder = path.join(__dirname, '../test Fixture with speci@l chars');
+
+	let driver: WebDriver;
+
+	before(async function() {
+		this.timeout(20000);
+		driver = VSBrowser.instance.driver;
+		await VSBrowser.instance.openResources(workspaceFolder);
+	});
+
+	after(async function() {
+		const editorView = new EditorView();
+		await editorView.closeAllEditors();
+	});
+
+	it('Open "empty.kaoto.yaml" file and check Kaoto UI is loading', async function () {
+		this.timeout(60000);
+		await VSBrowser.instance.openResources(path.join(workspaceFolder, 'empty.kaoto.yaml'));
+		const kaotoEditor = new CustomEditor();
+		assert.isFalse(await kaotoEditor.isDirty(), 'The Kaoto editor should not be dirty when opening it.');
+		const kaotoWebview = new WebView();
+		await driver.wait(async () => {
+			try {
+				await kaotoWebview.switchToFrame();
+				return true;
+			} catch {
+				return false;
+			}
+		}, 50000); // if it is the first load, it can take a lot of time as it downloads the docker image
+		await checkPartOfTopBarLoaded(driver);
+		await checkCanvasLoaded(driver);
+		await kaotoWebview.switchBack();
+		assert.isFalse(await kaotoEditor.isDirty(), 'The Kaoto editor should not be dirty after everything has loaded.');
+	});
+
+});
+
+async function checkCanvasLoaded(driver: WebDriver) {
+	await driver.wait(until.elementLocated(By.xpath("//div[text()='ADD A STEP']")));
```

Comment:
can we not use normal HTML selectors to check for the existence of something in the DOM, or is there a reason why we are using the driver + xpath? sorry I'm not too familiar with this approach so I may just need some enlightenment :)

Reply:
I used one way that is working. From my  far away experience several years ago), it was good habit to use either a `By.id` or a `By.xpath`. I'm open to use something else if it is better.

What is an HTML Selector?
The `By.name` https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_By.html ?

Reply:
An HTML selector, at least in this case, would be the other functions available, where you would select by the native HTML elements (e.g. by `id`, by `class`). I think if you are more comfortable with `By.xpath` then that should be fine too, it's simply that I am not used to it. :)